akka-http-core/src/main/scala/akka/http/engine/client/HostConnectionPoolGateway.scala

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.client
+
+import scala.annotation.tailrec
+import scala.concurrent.{ Future, Promise }
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+import scala.util.{ Failure, Try }
+import akka.actor.{ ActorLogging, Cancellable }
+import akka.stream.FlowMaterializer
+import akka.stream.actor.{ ActorPublisher, ActorSubscriber, ZeroRequestStrategy }
+import akka.stream.actor.ActorPublisherMessage._
+import akka.stream.actor.ActorSubscriberMessage._
+import akka.stream.impl.ResizableMultiReaderRingBuffer
```

Comment:
This use-case does neither resize nor have multiple readers, please use `akka.stream.impl.FixedSizeBuffer`.